File: libs/user-authentication/verification-code/src/main/kotlin/com/alesharik/user/authentication/verification/code/exception/VerificationSentTooFastException.java
package com.alesharik.user.authentication.verification.code.exception;

import org.zalando.problem.Status;
import org.zalando.problem.StatusType;
import org.zalando.problem.ThrowableProblem;

import java.net.URI;

public class VerificationSentTooFastException extends ThrowableProblem {
    public static final URI TYPE = URI.create("https://spring.alesharik.com/docs/error/auth/verification-sent-too-fast");

    @Override
    public URI getType() {
        return TYPE;
    }

    @Override
    public String getTitle() {
        return "Verification code sent request too fast";
    }

    @Override
    public String getDetail() {
        return "Slow down and wait before sending another request";
    }

    @Override
    public StatusType getStatus() {
        return Status.BAD_REQUEST;
    }
}

File: libs/user-authentication/verification-code/src/main/kotlin/com/alesharik/user/authentication/verification/code/exception/VerificationSentTooFastException.kt
package com.alesharik.user.authentication.verification.code.exception

import org.springframework.http.HttpStatus
import org.springframework.web.ErrorResponseException
import java.net.URI

class VerificationSentTooFastException : ErrorResponseException(HttpStatus.BAD_REQUEST) {
    init {
        setType(TYPE)
        setTitle("Verification code sent request too fast")
        setDetail("Slow down and wait before sending another request")
    }

    companion object {
        val TYPE = URI.create("https://spring.alesharik.com/docs/error/auth/verification-sent-too-fast")
    }
}
